### ISSUE TITLE
Health ingress uses old beta network api version (`extensions/v1beta1`)

### ISSUE THREAD

**Issue by christianacca** [bitnami/charts#8739]
**Which chart**:

aspnetcore 3.0.3

**Describe the bug**

The health ingress is using an old beta version of the ingress object. This is failing when deploying to a k8 v1.19+ cluster

**Expected behavior**

The [health-ingress.yaml](https://github.com/bitnami/charts/blob/master/bitnami/aspnet-core/templates/health-ingress.yaml) needs updating to auto-detect the api version of the networking capability - just like the ingress.yaml template already does. EG:

```yaml
apiVersion: {{ include "common.capabilities.ingress.apiVersion" . }}
```

It would also make sense to replace the use of annotations to define the ingress class, to instead use the standards field `ingressClassName`. EG:

```yaml
spec:
  {{- if and .Values.healthIngress.ingressClassName (include "common.ingress.supportsIngressClassname" .) }}
  ingressClassName: {{ .Values.healthIngress.ingressClassName | quote }}
  {{- end }}
```

**Comment by carrodher** [comment]
Thanks for the suggestion, would you like to contribute by creating a PR to solve the issue? The Bitnami team will be happy to review it and provide feedback.

**Comment by christianacca** [comment]
Yup I can do that :-)

**Comment by christianacca** [comment]
What's your thoughts on introducing `healthIngress.ingressClassName` as a replacement to a consumer of the chart providing `kubernetes.io/ingress.class: nginx` as an annotation?

Things that come to mind:
* that's a breaking change to the chart right? So requiring a bump of the major version
* should be a separate PR to the bug fix change that would introduce `apiVersion: {{ include "common.capabilities.ingress.apiVersion" . }}`
* I'm surprised that other charts in the repo haven't already adopted `healthIngress.ingressClassName`... maybe this should be a PR that fixes all of the health ingress templates in the repo (btw, I'm not sure I want to sign up for that work myself :-0 )?

**Comment by carrodher** [comment]
For the issue with the `apiVersion` we should use the capabilities present in the bitnami/common library chart. The `ingressClassName` should be present in the different charts containing an ingress object, but about `healthIngress`, as you rightly said, aspnet is the only chart including it.

About if it is a breaking change or not, I'm not sure, I would need to test if the backward compatibility is guaranteed, if not, it will require a major version bump for sure. If you want to split the changes into different PRs, it's totally fine to maintain the scope of each PR isolated.

Being said that, I'll ping my teammate @juan131 to see what is his opinion about the above questions

**Comment by christianacca** [comment]
I've created a PR for the immediate fix: #8794

I have another branch to introduce `healthIngress.ingressClassName`. Because `healthIngress.ingressClassName` is made optional, it's not going to be a breaking change. So I've bumped the minor version.

Once #8794 is merged to master, I'll rebase this other branch on top of master (to avoid the merge conflict to version field), and submit another PR
